sdk/src/gateway/client.ts

```diff
@@ -607,15 +591,181 @@ export class GatewayApiClient {
+                      | 'campaignId'
+                      | 'toToken'
+                      | 'maxSlippage'
+                      | 'fromChain'
+                  > & {
+                      toUserAddress: string;
+                      fromUserAddress: Address;
+                      fromChain: 'bob' | 'bob-sepolia';
+                      toChain: 'bitcoin';
+                  };
+              }
+            | {
+                  type: 'stake';
+                  params: StakeParams;
+              },
+        btcSigner: { signAllInputs: (psbtBase64: string) => Promise<string> },
+        walletClient: WalletClient<Transport, ViemChain, Account>,
+        publicClient: PublicClient<Transport>
+    ): Promise<string> {
+        if (order.type === 'onramp') {
+            const { uuid, psbtBase64 } = await this.startOrder(order.quote, order.params);
+
+            const bitcoinTxHex = await btcSigner.signAllInputs(psbtBase64!);
+
+            if (!bitcoinTxHex) throw new Error('no psbt');
+
+            const txId = await this.finalizeOrder(uuid, bitcoinTxHex);
+
+            return txId;
+        } else if (order.type === 'offramp') {
+            const tokenAddress = getTokenAddress(this.chainId, order.params.fromToken.toLowerCase());
+            const [offrampOrder, offrampRegistryAddress] = await Promise.all([
+                this.createOfframpOrder(order.params),
+                this.fetchOfframpRegistryAddress(),
+            ]);
+
+            const allowance = await publicClient.readContract({
+                address: tokenAddress,
+                abi: erc20Abi,
+                functionName: 'allowance',
+                args: [order.params.fromUserAddress as Address, offrampRegistryAddress],
+            });
+
+            if (BigInt(order.params.amount) > allowance) {
+                const { request } = await publicClient.simulateContract({
+                    account: walletClient.account,
+                    address: tokenAddress,
+                    abi: erc20Abi,
+                    functionName: 'approve',
+                    args: [offrampRegistryAddress, MaxUint256],
+                });
+
+                const approveTxHash = await walletClient.writeContract(request);
+
+                await publicClient.waitForTransactionReceipt({ hash: approveTxHash });
+            }
+
+            const { request } = await publicClient.simulateContract({
+                account: walletClient.account,
+                address: offrampRegistryAddress,
+                abi: offrampOrder.offrampABI,
+                functionName: offrampOrder.offrampFunctionName,
+                args: offrampOrder.offrampArgs,
+            });
+
+            const transactionHash = await walletClient.writeContract(request);
+
+            await publicClient?.waitForTransactionReceipt({ hash: transactionHash });
+
+            return transactionHash;
+        } else {
+            const result = await this.buildStake(order.params);
```

Comment:
After this, perform the allowance check & approve tx similar to how it's done in Offramp above, for the following code:

```ts
const { request } = await publicClient.simulateContract({
    account: walletClient.account,
    address: tokenAddress,
    abi: erc20Abi,
    functionName: 'approve',
    args: [order.strategyAddress, MaxUint256],
});
```